benchmarks/diffusion_2D/mpi_serial/diffusion.cpp

```diff
@@ -203,3 +200,260 @@ int laplacian(realtype t, N_Vector u, N_Vector f, void *user_data)
+  //   7. North Neighbor
+  //   8. East Neighbor
+
+  // Constants for computing Jacobian
+#if defined(BENCHMARK_ODE)
+  sunrealtype cx = udata->kx / (udata->dx * udata->dx);
+  sunrealtype cy = udata->ky / (udata->dy * udata->dy);
+  sunrealtype cc = -TWO * (cx + cy);
+#else
+  sunrealtype cx = -udata->kx / (udata->dx * udata->dx);
+  sunrealtype cy = -udata->ky / (udata->dy * udata->dy);
+  sunrealtype cc = cj - TWO * (cx + cy);
+#endif
+
+  // Value and Columns index
+  sunindextype idx = 0;
+
+  // west neighbor
+  if (i == 0)
+  {
+    // neighbor nx_loc
+    sunindextype nx_loc;
+    if (x - 1 < rx) nx_loc = qx + 1;
+    else nx_loc = qx;
+
+    vals[idx]    = cx;
+    col_idx[idx] = global_index(nx_loc - 1, j, x - 1, y, udata);
+    idx += 1;
+  }
+
+  // south neighbor
+  if (j == 0)
+  {
+    // neighbor ny_loc
+    sunindextype ny_loc;
+    if (y - 1 < ry) ny_loc = qy + 1;
+    else ny_loc = qy;
+
+    vals[idx]    = cy;
+    col_idx[idx] = global_index(i, ny_loc - 1, x, y - 1, udata);
+    idx += 1;
+  }
+
+  // south interior
+  if (j > 0)
+  {
+    vals[idx]    = cy;
+    col_idx[idx] = global_index(i, j - 1, x, y, udata);
+    idx += 1;
+  }
+
+  // west interior
+  if (i > 0)
+  {
+    vals[idx]    = cx;
+    col_idx[idx] = global_index(i - 1, j, x, y, udata);
+    idx += 1;
+  }
+
+  // center
+  vals[idx]    = cc;
+  col_idx[idx] = global_index(i, j, x, y, udata);
+  idx += 1;
+
+  // east interior
+  if (i < udata->nx_loc - 1)
+  {
+    vals[idx]    = cx;
+    col_idx[idx] = global_index(i + 1, j, x, y, udata);
+    idx += 1;
+  }
+
+  // north interior
+  if (j < udata->ny_loc - 1)
+  {
+    vals[idx]    = cy;
+    col_idx[idx] = global_index(i, j + 1, x, y, udata);
+    idx += 1;
+  }
+
+  // north neighbor
+  if (j == udata->ny_loc - 1)
+  {
+    vals[idx]    = cy;
+    col_idx[idx] = global_index(i, 0, x, y + 1, udata);
+    idx += 1;
+  }
+
+  // east neighbor
+  if (i == udata->nx_loc - 1)
+  {
+    vals[idx]    = cx;
+    col_idx[idx] = global_index(0, j, x + 1, y, udata);
+    idx += 1;
+  }
+
+  // non-zeros in this row
+  *row_nnz = 5;
+
+  return 0;
+}
+
+#if defined(BENCHMARK_ODE)
+int laplacian_matrix_sludist(N_Vector u, SUNMatrix L, UserData* udata)
+#else
+int laplacian_matrix_sludist(N_Vector u, sunrealtype cj, SUNMatrix L,
+                             UserData* udata)
+#endif
+{
+  SUNDIALS_CXX_MARK_FUNCTION(udata->prof);
+
+  // Set shortcuts
+  SuperMatrix* Lsuper    = SUNMatrix_SLUNRloc_SuperMatrix(L);
+  NRformat_loc* Lstore   = (NRformat_loc*)Lsuper->Store;
+  sunindextype* row_ptrs = Lstore->rowptr;
+  sunindextype* col_idxs = Lstore->colind;
+  sunrealtype* data      = (sunrealtype*)Lstore->nzval;
+
+  int x = udata->idx;
+  int y = udata->idy;
+
+  // Initial row pointer value
+  row_ptrs[0] = 0;
+
+  sunindextype idx     = 0;
+  sunindextype row_nnz = 0;
+
+  for (sunindextype j = 0; j < udata->ny_loc; j++)
+  {
+    for (sunindextype i = 0; i < udata->nx_loc; i++)
+    {
+#if defined(BENCHMARK_ODE)
```

Comment:
what is the point of this again?

Reply:
This is for the benchmark ARKODE/CVODE or IDA using the same function names but changing the signature for the ODE or DAE use case. When updating the benchmark I thought it would be better to just add `_ode` and `_dae` suffixes to the functions and get rid of `BENCHMARK_ODE` and `BENCHMARK_DAE` but decided to hold off on those changes for a different PR to keep this one more focused on the SuperLU additions.